opentracing-spring-jaeger-starter/src/main/java/io/opentracing/contrib/java/spring/jaeger/starter/log/slf4j/ScopeManagerSlf4jDecorator.java

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright 2018-2019 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.contrib.java.spring.jaeger.starter.log.slf4j;
+
+import io.jaegertracing.internal.JaegerSpanContext;
+import io.opentracing.Scope;
+import io.opentracing.ScopeManager;
+import io.opentracing.Span;
+import io.opentracing.contrib.java.spring.jaeger.starter.JaegerConfigurationProperties;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.MDC;
+
+
+/**
+ * @author <a href="mailto:taylor.tian@ericsson.com">Taylor Tian</a>
+ */
+public class ScopeManagerSlf4jDecorator implements ScopeManager {
+  private static final String B3_TRACE_ID = "X-B3-TraceId";

```

Comment:
Why do you need to distinguish B3 propagation? Cannot we just have two fields `traceid` and `spanid` ?

Reply:
I want to be compatible with B3 protocol or Spring Sleuth. The log config file won't need to be modified when we migrate legacy system from Sleuth to java-spring-jaeger if we keep compatible with B3.

Reply:
I think we should have a separate property to configure this.

One can use B3 with service mesh without using b3 names in the logs.

Reply:
Actually I use JaegerConfigurationProperties to configure this. Only when 'opentracing.jaeger.enableB3Propagation' is ture, those B3 key will be added to MDC.
`    if (properties.isEnableB3Propagation()) {
      currentContext.put(B3_TRACE_ID, context.toTraceId());
      currentContext.put(B3_SPAN_ID, context.toSpanId());
      currentContext.put(B3_SAMPLED, String.valueOf(context.isSampled()));
      currentContext.put(B3_EXPORT, String.valueOf(context.isSampled()));
    }`

Reply:
Yes another property.

Reply:
OK. I'll add a new properties class Slf4jJaegerProperties to configure these new features.